bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -926,6 +926,14 @@ impl Connection {
         })
     }
 
+    pub fn selected_application_protocol(&self) -> Result<Option<&str>, Error> {
+        let protocol = unsafe { s2n_get_application_protocol(self.connection.as_ptr()) };
+        if protocol.is_null() {
+            return Ok(None);
+        }
+        static_const_str!(protocol).map(|s| Some(s))
```

Comment:
I don't think it's safe to use `static_const_str` here, because while this data is const I don't think it's static, and it will only be valid for the lifetime of the connection.

I think this `&str` is still getting tied to the lifetime of the connection, but the name scares me 😅.

Reply:
I'm trying to clean up the naming in #4601.